src/styles/main.scss

```diff
@@ -1,3 +1,54 @@
+@import "./utils/variables";
+@import "./utils/functions";
+@import "./utils/resets";
+
 body {
+  line-height: 100px;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
   margin: 0;
 }
+
+.calendar {
+  display: flex;
+  margin: 150px auto;
+  flex-wrap: wrap;
+  gap: $gap;
+  width: width-calculate($day-size, $gap, $border-size);
+
+  @for $month-length from 28 through 31 {
+    &__month_length--#{$month-length} {
+      .calendar__day:nth-child(n + #{$month-length + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    height: $day-size;
+    width: $day-size;
+    box-sizing: border-box;
+    text-align: center;
+    background-color: #eee;
+    border: $border-size solid black;
+    transition: all 300ms ease-out;
```

Comment:
don't use all in your transitions, better specify all properties to make them refactorable in future